Assets/Scripts/Battlefield/TurnMechanism/TurnManager.cs

```diff
@@ -51,16 +55,17 @@ void Update()
 
             if (activePlayer.HasActionLeft())
             {
-                actionsLeft[0].color = new Color(0, 0, 1, .2f);
+                actionsLeft.color = new Color(0, 0, 1, .2f);
             }
             else
             {
-                actionsLeft[0].color = new Color(1, 0, 0, .2f);
+                actionsLeft.color = new Color(1, 0, 0, .2f);
             }
 
             if (activePlayer.creature.isEnemy)
             {
                 hotbar.enabled = false;
+                
             } else
             {
                 hotbar.enabled = true;
```

Comment:
This code can and should be abstracted out into a component instead of seeing it here in the TurnManger. A Manager should NOT contain and real code or logic but instead composes a system, so it would have a reference to an object that can deal with indicating number of actions like another MonoBehavior component for a UI or a non MonoBehavior class that has it's update called from TurnManagers update.

Reply:
I only made this small change because i was looking over this section.  This whole actionsLeft system is going to change and be handled when we do proper ui management.  The changes you are requesting here will happen, but not in this branch.  A seperate issue and branch show be made for this